File: CS/DevExpressMvcApplication1/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace DevExpressMvcApplication1.Controllers {
	public class HomeController : Controller {
		public ActionResult Index() {
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

			return View(NorthwindDataProvider.GetEmployees());
		}
		public ActionResult InlineEditingPartial() {
			return PartialView("GridView", NorthwindDataProvider.GetEmployees());
		}
		public ActionResult HtmlEditorPartial(string employeeID) {			
			int empId= Int32.Parse(employeeID);
			Employee emp = empId> 0 
				? emp = NorthwindDataProvider.GetEmployee(empId)
				: new Employee();
			
			return PartialView("HtmlEditor",  emp);
		}		
		[HttpPost]
		public ActionResult InlineEditingAddNewPartial([ModelBinder(typeof(DevExpressEditorsBinder))] Employee employee) {
                                                employee.Notes = HtmlEditorExtension.GetHtml("heFeatures");
			if (ModelState.IsValid) {
				try {
					//update a database
				}
				catch (Exception e) {
					ViewData["EditError"] = e.Message;
				}
			}
			else
				ViewData["EditError"] = "Please, correct all errors.";
			return PartialView("GridView", NorthwindDataProvider.GetEmployees());
		}
		[HttpPost]
		public ActionResult InlineEditingUpdatePartial([ModelBinder(typeof(DevExpressEditorsBinder))] Employee employee) {
                                                employee.Notes = HtmlEditorExtension.GetHtml("heFeatures");
			if (ModelState.IsValid) {
				try {
					//update a database
				}
				catch (Exception e) {
					ViewData["EditError"] = e.Message;
				}
			}
			else
				ViewData["EditError"] = "Please, correct all errors.";

			return PartialView("GridView", NorthwindDataProvider.GetEmployees());
		}
		[HttpPost]
		public ActionResult InlineEditingDeletePartial(int employeeID) {
			if (employeeID > 0) {
				try {
					//update a database
				}
				catch (Exception e) {
					ViewData["EditError"] = e.Message;
				}
			}
			return PartialView("GridView", NorthwindDataProvider.GetEmployees());
		}
	}
}

File: VB/DevExpressMvcApplication1/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpressMvcApplication1.Models

Namespace DevExpressMvcApplication1.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View(Repository.Persons)
		End Function
		Public Function InlineEditingPartial() As ActionResult
			Return PartialView("GridView", Repository.Persons)
		End Function
		Public Function HtmlEditorPartial(ByVal personID As String) As ActionResult
			Dim persID As Integer = Int32.Parse(personID)
			Dim pers As Person
			If persID > 0 Then
				pers = Repository.GetPersonByID(persID)
			Else
				pers = New Person()
			End If

			Return PartialView("HtmlEditor", pers)
		End Function
		<HttpPost>
		Public Function InlineEditingAddNewPartial(ByVal person As Person) As ActionResult
			If ModelState.IsValid Then
				Try
					Repository.InsertPerson(person)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			Else
                ViewData("EditError") = "Please, correct all errors."
                ViewData("DescriptionValue") = person.Description
            End If
			Return PartialView("GridView", Repository.Persons)
		End Function
		<HttpPost>
		Public Function InlineEditingUpdatePartial(ByVal person As Person) As ActionResult
			If ModelState.IsValid Then
				Try
					Repository.UpdatePerson(person)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			Else
                ViewData("EditError") = "Please, correct all errors."
                ViewData("DescriptionValue") = person.Description
            End If

			Return PartialView("GridView", Repository.Persons)
		End Function
		<HttpPost>
		Public Function InlineEditingDeletePartial(ByVal id As Integer) As ActionResult
			If id > 0 Then
				Try
					Repository.RemovePersonByID(id)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			End If
			Return PartialView("GridView", Repository.Persons)
		End Function
	End Class
End Namespace